weldx/core.py

```diff
@@ -459,20 +475,14 @@ def _interp_time_discrete(self, time: Time) -> xr.DataArray:
     def _interp_time_expression(self, time: Time, time_unit: str) -> xr.DataArray:
         """Interpolate the time series if its data is a mathematical expression."""
         time_q = time.as_quantity(unit=time_unit)
+        if len(time_q.shape) == 0:
+            time_q = np.expand_dims(time_q, 0)
 
-        if len(self.shape) > 1 and np.iterable(time_q):
-            while len(time_q.shape) < len(self.shape):
-                time_q = time_q[:, np.newaxis]
+        time_xr = xr.DataArray(time_q, dims=["time"])
 
         # evaluate expression
-        data = self._data.evaluate(**{self._time_var_name: time_q})
-        data = data.astype(float).to_reduced_units()  # float conversion before reduce!
-
-        # create data array
-        if not np.iterable(data):  # make sure quantity is not scalar value
-            data = np.expand_dims(data, 0)
-
-        return self._create_data_array(data, time)
+        data = self._data.evaluate(**{self._time_var_name: time_xr})
+        return data.assign_coords({"time": time.as_timedelta_index()})
```

Comment:
I guess we don't really support reference times with `TimeSeries` using ME?
Still I think we should keep that information whenever possible to make future implementations easier

Reply:
Agreed. I just tried to change that, but unfortunately, some tests are failing because of "non-monotonic" indices and I couldn't figure out why by just looking at it. Maybe we should make an issue and do this in another PR. At least I got it working for the ME at the position you marked above.

Reply:
the non-monotonic error surprises me, do we have a CI run with that popping up?

Reply:
no, but I just realized, that I made an error. So it might not be an issue.